python/sglang/lang/backend/runtime_endpoint.py

```diff
@@ -216,21 +220,15 @@ def select(
         s: StreamExecutor,
         choices: List[str],
         temperature: float,
-    ):
+        choices_method: Optional[ChoicesSamplingMethod],
+    ) -> ChoicesDecision:
         assert temperature <= 1e-5
+        choices_method = choices_method or token_length_normalized
```

Comment:
The `RuntimeEndpoint` is expected to execute something very clear and specific, so please do not resolve the default argument here. Move this line to somewhere earlier in the call stack, such as `api.py`